docs/inference-providers/guides/evaluation-inspect-ai.md

```diff
@@ -0,0 +1,258 @@
+        solver=generate(),
+        scorer=model_graded_fact()
+    )
+
+target_models = [
+    "MiniMaxAI/MiniMax-M2",
+    "openai/gpt-oss-20b",
+    "openai/gpt-oss-120b", 
+    "moonshotai/Kimi-K2-Instruct-0905"
+]
+for model in target_models:
+    eval(theory_of_mind(), model=f"hf-inference-providers/{model}")
+```
+
+If we save this to a file `models_bench.py` we can run it like this:
+
+```bash
+python models_bench.py
+```
+
+If everything went well we will see the evaluation running sequentially for each model on the list. To analyze the results, we can use the viewer:
+
+
+```bash
+inspect viewer
+```
+![Screenshot of inspect viewer results with gpt-oss-20b](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers-guides/evals-guide-model-bench-viewer.png).
+
+
+## Example: Comparing several inference providers for a task
+In this section, we will evaluate the same model across different providers. Inference Providers gives us access to many providers for the same model. As performance might vary across providers, this is a useful metric to select the provider, in addition to other important factors such as speed and cost.
+
+If we don't specify a provider, like we did in our previous examples, the system automatically routes your request to the first available provider for the specified model, following your preference order in [Inference Provider settings](https://hf.co/settings/inference-providers). But we can also select the provider by appending the provider name to the model id (e.g. `openai/gpt-oss-120b:sambanova`).
+
+Let's run the evaluations for `gpt-oss-120b` across several providers:
+
+```python
+from inspect_ai import eval
+from inspect_ai import Task, task
+from inspect_ai.dataset import example_dataset
+from inspect_ai.scorer import model_graded_fact
+from inspect_ai.solver import generate
+
+@task
+def theory_of_mind():
+    return Task(
+        dataset=example_dataset("theory_of_mind"),
+        solver=generate(),
+        scorer=model_graded_fact()
+    )
+
+target_providers = [
+    "together",
+    "sambanova", 
+    "groq",
+    "novita",
+    "nebius",
+    "cerebras",
+    "nscale",
+    "hyperbolic",
+    "fireworks-ai",
+    "scaleway"
+]
+for provider in target_providers:
+    eval(theory_of_mind(), model=f"hf-inference-providers/openai/gpt-oss-120b:{provider}")
+```
+
+Launching the viewer and sorting by score, we can compare the performance across providers, similar to the table below:
+
+| Model | Provider | Score |
+|-------|----------|-------|
+| openai/gpt-oss-120b | hyperbolic | 0.84 |
+| openai/gpt-oss-120b | fireworks-ai | 0.82 |
+| openai/gpt-oss-120b | nscale | 0.82 |
+| openai/gpt-oss-120b | nebius | 0.82 |
+| openai/gpt-oss-120b | scaleway | 0.81 |
+| openai/gpt-oss-120b | together | 0.81 |
+| openai/gpt-oss-120b | cerebras | 0.8 |
+| openai/gpt-oss-120b | novita | 0.8 |
+| openai/gpt-oss-120b | groq | 0.8 |
+| openai/gpt-oss-120b | sambanova | 0.8 |
+
```

Comment:
Maybe worth a quick note to say you can also filter using `:cheapest` and `:fastest` https://github.com/huggingface/hub-docs/pull/2025/